### DISCUSSION TITLE
Add link(user drag) start from target?

### DISCUSSION THREAD

**Discussion by zta7** [clientIO/joint#2081] [Q&A] [answered]
### Introduction

I want to add a link from target not source by user dragging. 
There are no good options in the documentation, maybe adding source/target options in the source code is a good option?
Or any other solution can help?
![image](https://user-images.githubusercontent.com/30122066/226267116-03080f46-072c-49f5-ac12-cf48b4c8f671.png)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment, marked as answer]
There is no API to do it right now. What's the use case?

What you can do is to create a temporary link (perhaps with arrow at the source) and replace it with a new link when the user connects an element.

**Reply by zta7** [reply]
In my application, all ports are clearly define as "in-port(target)" or "output-port(source)". 
Just like https://www.jointjs.com/demos/working-with-ports
My purpose is that i want users to be free drag from the target or the source not only can drag start with the source.
It's not about exchange the source and target, i want the magnet be marked as target and start link from link.target.

**Reply by kumilingus** [reply]
Then you should change the [magnet](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#dia.attributes.magnet) attribute from `"passive"` to `"true"`.

**Reply by zta7** [reply]
But it's treat in-port as source which should be the target ....

**Reply by kumilingus** [reply]
Sorry, I don't understand. What does it mean to be treated as source / target in your point of view?

In JointJS, every link has `source` and `target` end. If you want make sure that the link's `target` end is at the `in-port` you can switch the direction of the link as I suggested above.

**Reply by kumilingus** [reply]
Please try to create a JSFiddle that demonstrates the problem.

**Reply by zta7** [reply]
Drag a link like this ~
![image](https://user-images.githubusercontent.com/30122066/226600660-e622ab1a-5d8a-44ab-9ee8-3604ea5399f1.png)

**Reply by kumilingus** [reply]
Yes, you need to approach this with a workaround I suggested earlier.

> What you can do is to create a temporary link (perhaps with arrow at the source) and replace it with a new link when the user connects an element.

**Reply by kumilingus** [reply]
Here's an example: https://jsfiddle.net/kumilingus/3r2mwh9q/
